### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by kagikn**
I'd like to improve performance of `ObjectFromNative` for value types too, but I'm not sure how to do without duplicating certain code.

### CODE REVIEW

File: source/core/Native.hpp [hunk 1]
```diff
@@ -19,6 +19,8 @@
 #include "Model.hpp"
 #include "NativeHashes.hpp"
 
+using namespace System;
```
Comment (crosire) [inline source/core/Native.hpp]: This doesn't have any effect on the current code below, so seems unnecessary.

**Comment by kagikn** [comment]
Looks like there was significant performance drop where the methods pushed arguments to natives and retrieved return values from them without avoiding boxing/unboxing of primitive types.

So I'll improve the performance of the code to get results from natives as soon as possible.

File: source/core/Native.hpp [hunk 2]
```diff
@@ -212,5 +215,46 @@ namespace GTA
 			generic <typename T>
 			static T Call(System::UInt64 hash, ... array<InputArgument ^> ^arguments);
 		};
+
+		private ref class NativeHelper abstract sealed
```
Comment (crosire) [inline source/core/Native.hpp]: These should probably go into Native.cpp, since they aren't used anywhere else.

**Comment by kagikn** [comment]
Another major problem of the performance must be making one thread for every scripts (See #836), but it'll be difficult to solve. So would you check if it's okay to merge this PR? @crosire

**Comment by crosire** [comment]
This is some cool stuff. Don't have the time to test it in-game right now, but it looks good code-wise, so should be okay to merge.

**Pull request merged**